sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java

```diff
@@ -122,15 +122,34 @@
    * pattern).
    */
   public static Read read(Schema schema) {
-    return new AutoValue_ParquetIO_Read.Builder().setSchema(schema).build();
+    return new AutoValue_ParquetIO_Read.Builder()
+        .setSchema(schema)
+        .setInferBeamSchema(false)
+        .build();
   }
 
   /**
    * Like {@link #read(Schema)}, but reads each file in a {@link PCollection} of {@link
    * org.apache.beam.sdk.io.FileIO.ReadableFile}, which allows more flexible usage.
    */
   public static ReadFiles readFiles(Schema schema) {
-    return new AutoValue_ParquetIO_ReadFiles.Builder().setSchema(schema).build();
+    return new AutoValue_ParquetIO_ReadFiles.Builder()
+        .setSchema(schema)
+        .setInferBeamSchema(false)
+        .build();
+  }
+
+  private static <T> PCollection<T> setBeamSchema(
+      PCollection<T> pc, Class<T> clazz, @Nullable Schema schema) {
+    org.apache.beam.sdk.schemas.Schema beamSchema =
+        org.apache.beam.sdk.schemas.utils.AvroUtils.getSchema(clazz, schema);
+    if (beamSchema != null) {
```

Comment:
Because here you won't throw exception if there is not beamSchema, why not just make "inforSchema" as a non-optional action so we don't need set the boolean to control it?

Reply:
Thanks for reviewing the code.  
The idea here was to align it the same way as AvroIO (https://github.com/apache/beam/blob/ad5d3836a47fe2cbd552fe3908e15ffc7f777f11/sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java#L339) 
Currently AvroUtils.getSchema() returns null if it is anything other than GenericRecord type. 
Currently In parquetIO we are handling only generic record so this method will always return schema. 
I think with your suggestion of making infer schema as non optional and as we handle only GenericRecord in ParquetIO, I will modify the code to set the beamSchema always without the boolean flag as we do in AvroIO.

Reply:
@amaliujia I have done the modification. Can you have a look